diff --git a/interface/INetworkManager.h b/interface/INetworkManager.h
index 65c982a..55f1cfc 100644
--- a/interface/INetworkManager.h
+++ b/interface/INetworkManager.h
@@ -258,7 +258,7 @@ namespace WPEFramework
             virtual uint32_t GetLogLevel(Logging& level /* @out */) = 0;
 
             /* @brief Configure the Network Manager plugin */
-            virtual uint32_t Configure(PluginHost::IShell* service /* @in */) = 0;
+            virtual uint32_t Configure(const string configLine/* @in */) = 0;
 
             /* @event */
             struct EXTERNAL INotification : virtual public Core::IUnknown
diff --git a/plugin/NetworkManager.cpp b/plugin/NetworkManager.cpp
index 250b3e0..1049e50 100644
--- a/plugin/NetworkManager.cpp
+++ b/plugin/NetworkManager.cpp
@@ -95,18 +95,13 @@ namespace WPEFramework
                 _networkManager->Register(&_notification);
 
                 SYSLOG(Logging::Startup, (_T("Configuring NetworkManager")));
-                Exchange::IConfiguration* config = _networkManager->QueryInterface<Exchange::IConfiguration>();
-                if (config != nullptr) {
-                    if(config->Configure(service) != Core::ERROR_NONE)
-                    {
-                        SYSLOG(Logging::Shutdown, (_T("NetworkManager failed to configure")));
-                        message = _T("NetworkManager failed to configure");
-                    }
-                    else
-                    {
-                        SYSLOG(Logging::Startup, (_T("Configuring successful")));
-                    }
-                    config->Release();
+                if (_networkManager->Configure(service->ConfigLine()) != Core::ERROR_NONE)
+                {
+                    SYSLOG(Logging::Shutdown, (_T("NetworkManager failed to configure")));
+                }
+                else
+                {
+                    SYSLOG(Logging::Startup, (_T("Configuring successful")));
                 }
                 // Register all custom JSON-RPC methods
                 SYSLOG(Logging::Startup, (_T("Registering JSONRPC Methods")));
diff --git a/plugin/NetworkManager.h b/plugin/NetworkManager.h
index 98007e8..3c09e5a 100644
--- a/plugin/NetworkManager.h
+++ b/plugin/NetworkManager.h
@@ -22,7 +22,6 @@
 #include "Module.h"
 
 #include "INetworkManager.h"
-#include <interfaces/IConfiguration.h>
 #include "NetworkManagerLogger.h"
 
 #include <string>
diff --git a/plugin/NetworkManagerImplementation.cpp b/plugin/NetworkManagerImplementation.cpp
index 521e871..4bfc1ff 100644
--- a/plugin/NetworkManagerImplementation.cpp
+++ b/plugin/NetworkManagerImplementation.cpp
@@ -116,29 +116,19 @@ namespace WPEFramework
             return Core::ERROR_NONE;
         }
 
-        uint32_t NetworkManagerImplementation::Configure(PluginHost::IShell* service)
+        uint32_t NetworkManagerImplementation::Configure(const string configLine)
         {
             LOG_ENTRY_FUNCTION();
             Configuration config;
-            if (service)
+            if(configLine.empty())
             {
-                string configLine = service->ConfigLine();
-                if(configLine.empty())
-                {
-                    NMLOG_FATAL("config line : is empty !");
-                    return Core::ERROR_GENERAL;
-                }
-                else
-                {
-                    NMLOG_INFO("Loading the incoming configuration : %s", configLine.c_str());
-                    config.FromString(configLine);
-                }
+                NMLOG_FATAL("config line : is empty !");
+                return Core::ERROR_GENERAL;
             }
             else
             {
-                NMLOG_FATAL("Service is NULL!");
-                return Core::ERROR_GENERAL;
-
+                NMLOG_INFO("Loading the incoming configuration : %s", configLine.c_str());
+                config.FromString(configLine);
             }
 
             NetworkManagerLogger::SetLevel(static_cast <NetworkManagerLogger::LogLevel>(config.loglevel.Value()));
diff --git a/plugin/NetworkManagerImplementation.h b/plugin/NetworkManagerImplementation.h
index d6077e8..3b62f82 100644
--- a/plugin/NetworkManagerImplementation.h
+++ b/plugin/NetworkManagerImplementation.h
@@ -30,7 +30,6 @@
 using namespace std;
 
 #include "INetworkManager.h"
-#include <interfaces/IConfiguration.h>
 #include "NetworkManagerLogger.h"
 #include "NetworkManagerConnectivity.h"
 #include "NetworkManagerStunClient.h"
@@ -50,7 +49,7 @@ namespace WPEFramework
 {
     namespace Plugin
     {
-        class NetworkManagerImplementation : public Exchange::INetworkManager, public Exchange::IConfiguration
+        class NetworkManagerImplementation : public Exchange::INetworkManager
         {
             enum NetworkEvents
             {
@@ -178,7 +177,6 @@ namespace WPEFramework
 
                 BEGIN_INTERFACE_MAP(NetworkManagerImplementation)
                 INTERFACE_ENTRY(Exchange::INetworkManager)
-                INTERFACE_ENTRY(Exchange::IConfiguration)
                 END_INTERFACE_MAP
 
                 // Handle Notification registration/removal
@@ -250,7 +248,7 @@ namespace WPEFramework
                 uint32_t GetLogLevel(Logging& level /* @out */) override;
 
                 /* @brief configure network manager plugin */
-                uint32_t Configure(PluginHost::IShell* service) override;
+                uint32_t Configure(const string configLine) override;
 
                 /* Events */
                 void ReportInterfaceStateChange(const Exchange::INetworkManager::InterfaceState state, const string interface);
