diff --git a/plugin/gnome/NetworkManagerGnomeEvents.cpp b/plugin/gnome/NetworkManagerGnomeEvents.cpp
index 413a0aa..fa65147 100644
--- a/plugin/gnome/NetworkManagerGnomeEvents.cpp
+++ b/plugin/gnome/NetworkManagerGnomeEvents.cpp
@@ -24,7 +24,7 @@
 #include <string>
 #include <map>
 #include <NetworkManager.h>
-#include <libnm/NetworkManager.h>
+//#include <libnm/NetworkManager.h>
 #include "Module.h"
 #include "NetworkManagerGnomeEvents.h"
 #include "NetworkManagerLogger.h"
diff --git a/plugin/gnome/NetworkManagerGnomeProxy.cpp b/plugin/gnome/NetworkManagerGnomeProxy.cpp
index 76e7eff..bd6f9e3 100644
--- a/plugin/gnome/NetworkManagerGnomeProxy.cpp
+++ b/plugin/gnome/NetworkManagerGnomeProxy.cpp
@@ -20,7 +20,7 @@
 #include "NetworkManagerGnomeWIFI.h"
 #include "NetworkManagerGnomeEvents.h"
 #include "NetworkManagerGnomeUtils.h"
-#include <libnm/NetworkManager.h>
+//#include <libnm/NetworkManager.h>
 #include <fstream>
 #include <sstream>
 
diff --git a/plugin/gnome/NetworkManagerGnomeUtils.cpp b/plugin/gnome/NetworkManagerGnomeUtils.cpp
index a6b2bdb..8189d8f 100644
--- a/plugin/gnome/NetworkManagerGnomeUtils.cpp
+++ b/plugin/gnome/NetworkManagerGnomeUtils.cpp
@@ -35,8 +35,8 @@ namespace WPEFramework
 {
     namespace Plugin
     {
-        static std::string m_ethifname = "eth0";
-        static std::string m_wlanifname = "wlan0";
+        static std::string m_ethifname = "eno1";
+        static std::string m_wlanifname = "wlp2s0";
 
         const char* nmUtils::wlanIface() {return m_wlanifname.c_str();}
         const char* nmUtils::ethIface() {return m_ethifname.c_str();}
diff --git a/plugin/gnome/NetworkManagerGnomeWIFI.cpp b/plugin/gnome/NetworkManagerGnomeWIFI.cpp
index 602b942..b429804 100644
--- a/plugin/gnome/NetworkManagerGnomeWIFI.cpp
+++ b/plugin/gnome/NetworkManagerGnomeWIFI.cpp
@@ -50,10 +50,21 @@ namespace WPEFramework
         {
             GError *error = NULL;
             if(m_client != nullptr)
+            {
+                GMainContext *context = g_main_context_ref(nm_client_get_main_context(m_client));
+                GObject *context_busy_watcher = nm_client_get_context_busy_watcher(m_client);
+                g_object_add_weak_pointer(context_busy_watcher,(gpointer *) &context_busy_watcher);
+                g_clear_object(&m_client);
+                while (context_busy_watcher)
+                {
+                    g_main_context_iteration(context, TRUE);
+                }                                                                                                                         g_main_context_unref(context);
+            }
+            /*if(m_client != nullptr)
             {
                 g_object_unref(m_client);
                 m_client = nullptr;
-            }
+            }*/
 
             m_client = nm_client_new(NULL, &error);
             if (!m_client || !m_loop) {
diff --git a/plugin/gnome/NetworkManagerGnomeWIFI.h b/plugin/gnome/NetworkManagerGnomeWIFI.h
index 0e95dfe..e33b0f9 100644
--- a/plugin/gnome/NetworkManagerGnomeWIFI.h
+++ b/plugin/gnome/NetworkManagerGnomeWIFI.h
@@ -72,7 +72,11 @@ namespace WPEFramework
             wifiManager();
             ~wifiManager() {
                 NMLOG_INFO("~wifiManager");
-                g_main_context_pop_thread_default(m_nmContext);
+                if (m_nmContext) {
+                    g_main_context_pop_thread_default(m_nmContext);
+                    g_main_context_unref(m_nmContext);
+                    m_nmContext = NULL;
+                }
                 if(m_client != NULL) {
                     g_object_unref(m_client);
                     m_client = NULL;
