diff --git a/Source/websocket/JSONRPCLink.h b/Source/websocket/JSONRPCLink.h
index 9652d7ec..aad19ab1 100644
--- a/Source/websocket/JSONRPCLink.h
+++ b/Source/websocket/JSONRPCLink.h
@@ -1235,16 +1235,6 @@ namespace WPEFramework {
 				}
 
 			public:
-				template <typename INBOUND, typename METHOD>
-				uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method)
-				{
-					return Subscribe<INBOUND, METHOD>(waitTime, eventName, method);
-				}
-				template <typename INBOUND, typename METHOD, typename REALOBJECT>
-				uint32_t Subscribe(const uint32_t waitTime, const string& eventName, const METHOD& method, REALOBJECT* objectPtr)
-				{
-					return Subscribe<INBOUND, METHOD, REALOBJECT>(waitTime, eventName, method, objectPtr);
-				}
 				bool IsActivated()
 				{
 					return (_state == ACTIVATED);
